packages/hooks/src/useLoggedStateLoginAndLogout.ts

```diff
@@ -0,0 +1,80 @@
+import { useEffect } from 'react';
+import Cookies from 'js-cookie';
+import { isSafari } from '@deriv/shared';
+import { requestOidcAuthentication } from '@deriv-com/auth-client';
+
+/**
+ * Handles silent login and single logout logic for OAuth2.
+ *
+ * @param {{
+ *   is_client_store_initialized: boolean; // Whether the client store has been initialized
+ *   isOAuth2Enabled: boolean; // Whether OAuth2 feature is enabled
+ *   oAuthLogout: () => Promise<void>; // Function to handle OAuth2 logout
+ * }} params - The arguments required for silent login and logout management
+ */
+const useLoggedStateLoginAndLogout = ({
+    is_client_store_initialized,
```

Comment:
This is the main hook for the `logged_state` logic, I just renamed it to `useLoggedStateLoginAndLogout`